### PR TITLE
Remove tensorflow-metadata folders

### PR DESCRIPTION
**Pull request by casassg**
Instead of sourcing tensorflow-metadata files in the sdk, we add a dependency on tensorflow-metadata library.

Currently feast python adds tensorflow_metadata namespace. This causes pip to overwrite the tensorflow-metadata library if installed after. 

So instead of doing this, we can just add a dependency on tensorflow-metadata.

This PR is a proposal to fix the issue detected in #2035

```release-note
Switch to use tensorflow-metadata requirement instead of sourcing compiled tensorflow-metadata in the sdk.
```

### CODE REVIEW

**Comment by woop** [comment]
Thanks @casassg!

File: sdk/python/setup.py [hunk 1]
```diff
@@ -65,6 +65,7 @@
     "fastapi>=0.68.0",
     "uvicorn[standard]>=0.14.0",
     "proto-plus<1.19.7",
+    "tensorflow-metadata>=1.0.0<2.0.0",
```
Comment (achals) [inline sdk/python/setup.py:68]: I think you'll need a comma here between `>=1.0.0` and `<2.0.0`, right?

Comment (casassg) [reply sdk/python/setup.py]: you are correct, ty for seeing this. Totally slipped me.

**Comment by casassg** [comment]
(rebased for signoff)

**Review by achals — APPROVED** [review]
/lgtm

**Comment by casassg** [comment]
(is there a schedule for when this may be released? Mostly want to use this for tfx-addons project :D )

**Comment by achals** [comment]
@casassg @adchia can comment but I believe we're planning on cutting a new release in a the coming couple of weeks.

**Comment by casassg** [comment]
Sg! Will wait for confirmation. Would be nice otherwise if we can backport this to 0.15 as a patch release given that this is mostly a "bug" fix (I understand it's not, but it should not modify behaviour as it effectively should be the same)

**Comment by adchia** [comment]
There's likely some patch release we could sneak in relatively soon since we do have several issues we want to resolve.

**Pull request merged**